main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -509,6 +510,63 @@ object CompletionProvider {
     }
   }
 
+  private def caseCompletion(label: String, completion: String)(implicit context: Context): CompletionItem = {
+    CompletionItem(label = label,
+        sortText = Priority.high(label),
+        textEdit = TextEdit(context.range, completion),
+        documentation = None,
+        insertTextFormat = InsertTextFormat.Snippet,
+        kind = CompletionItemKind.EnumMember)
+  }
+
+  private def enumCompletionAcc(acc: List[CompletionItem], enm: TypedAst.Enum, currentWordIsCase: Boolean)(implicit context: Context, flix: Flix): List[CompletionItem] = {
+    enm.cases.foldLeft(acc)({
+      case (acc, (sym , cas)) => {
+        val name = sym.name
+        val tpe = cas.tpe
+        val typeString = FormatType.formatType(tpe) match {
+          case "Unit" => ""
+          case x if x.charAt(0) == '(' => x
+          case x => s"($x)"
+        }
+        val typeCompletion = tpe.typeConstructor match {
+          case Some(TypeConstructor.Unit) => ""
+          case Some(TypeConstructor.Tuple(arity)) => List.range(1, arity + 1).map(elem => s"$$$elem").mkString("(", ", ", ")")
+          case _ => "($1)"
+        }
+        val label = if (currentWordIsCase) s"case $name$typeString => " else s"$name$typeString => "
+        val completion = if (currentWordIsCase) s"case $name$typeCompletion => $${0:???}" else s"$name$typeCompletion => $${0:???}"
+        caseCompletion(label, completion) :: acc
+      }
+    })
+  }
+
+  /** 
+    * Returns a list of completion items based on case keyword in match expressions
+    */
+  private def getCaseCompletions()(implicit context: Context, index: Index, root: TypedAst.Root, flix: Flix): Iterable[CompletionItem] = {
```

Comment:
We usually list functions in this order:

- The function
- A helper function
- A helper function

so I think getCaseCompletions should be first followed by its helpers.